docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,292 @@
+    0,
+    NULL, // opt_params
+    0,
+    &report,
+    report_size);
+
+result = oe_get_collaterals(
+    &collaterals,
+    &collaterals_size);
+...
+```
+
+##### Challenger verifies the evidence and collaterals (in the untrusted host)
+```C
+...
+
+// Verify report with collaterals
+result = oe_verify_remote_report_with_collaterals(
+            report,
+            report_size,
+            collaterals,
+            collaterals_size,
+            NULL, // input_validation_time
+            NULL, // parsed_report);
+...
+```
+
+### 3. Challenger specifies collaterals:
+In this scenario the attester only provides the evidence to the challenger.  The challenger then fetches the collaterals from a second source different than the OE SDK, and uses the evidence and collaterals to verify the TEE.
+
+##### Attester generates the evidence and collaterals (inside an enclave/TEE)
+```C
+...
+result = oe_get_report(
+    OE_REPORT_FLAGS_REMOTE_ATTESTATION,
+    NULL, // report_data
+    0,
+    NULL, // opt_params
+    0,
+    &report,
+    report_size);
+...
+```
+
+##### Challenger verifies the evidence with custom collaterals (in the untrusted host)
+```C
+...
+//
+// Challenger gets collaterals not using OE SDK
+//
+
+//
+// Challenger builds **collaterals** structure
+//
+
+// Verify report with collaterals
+result = oe_verify_remote_report_with_collaterals(
+            report,
+            report_size,
+            collaterals,
+            collaterals_size,
+            NULL, // input_validation_time
+            NULL, // parsed_report);
+...
+```
+
+Specification
+-------------
+
+### Collateral structure
+Generic serializable structure that stores the collaterals in raw binary format.
+
+`report.h`
+```C
+/*! \struct oe_collaterals_t
+ *
+ * \brief OE collaterals
+ *
+ * Raw generic serializable structure that contains the collaterals.
+ *
+ */
+typedef struct _oe_collaterals_t
+{
+    uint32_t version;
+    uint8_t enclave_type;         ///< The type of enclave
+    uint8_t reserved;
+    uint16_t reserved2;
+    uint32_t num_elements;
+    size_t buffer_size;           ///< Size of the buffer
+
+    ///< Data buffer is made of an offset array of type uint32_t, followed by
+    ///< the actual data.
+    ///< This array has the size of **num_elements** and stores the offset
+    ///< into the data section.
+    ///<  _________________________
+    ///<  |  version              |
+    ///<  |-----------------------|
+    ///<  |  reserved    | type   |
+    ///<  |-----------------------|
+    ///<  |  num_elements         |
+    ///<  |-----------------------|
+    ///<  |  buffer_size          |
+    ///<  |-----------------------|
+    ///<  |                       |
+    ///<  |  array(uint32_t)      |
+    ///<  |  size of num_elements |
+    ///<  |-----------------------|
+    ///<  |  Data                 |
+    ///<  |_______________________|
+    ///<
+    uint8_t buffer[];              ///< Buffer of offsets + data
```

Comment:
Line 163 says array of uint32_t but here it's uint8_t.   Can this be uint32_t?

Reply:
I am just trying to state is a raw buffer of bytes.  I rather deal with bytes then 4 bytes words.